result.go

```diff
@@ -479,12 +483,17 @@ func newResultGrouped(f reflect.StructField) (resultGrouped, error) {
 	return rg, nil
 }
 
-func (rt resultGrouped) Extract(cw containerWriter, v reflect.Value) {
-	if !rt.Flatten {
+func (rt resultGrouped) Extract(cw containerWriter, decorated bool, v reflect.Value) {
+	// Decorated values are always flattened.
+	if !decorated && !rt.Flatten {
 		cw.submitGroupedValue(rt.Group, rt.Type, v)
 		return
 	}
 	for i := 0; i < v.Len(); i++ {
-		cw.submitGroupedValue(rt.Group, rt.Type, v.Index(i))
+		if !decorated {
+			cw.submitGroupedValue(rt.Group, rt.Type, v.Index(i))
+		} else {
+			cw.submitDecoratedGroupedValue(rt.Group, rt.Type.Elem(), v.Index(i))
+		}
```

Comment:
I think maybe this is where—if we're decorating—instead of submitting individual values, we submit the entire decorated slice, so that scope level knows whether it decorated that group.

This will allow replacing the decorated value groups wholesale.